### PR TITLE
Report reminders

### LINKED ISSUE #3989: Configurable project report reminders
**Is your feature request related to a problem? Please describe.**

Staff would rather not set up reminders on each and every submission but configure it more generally.

**Is your feature request related to an existing functionality? Please describe.**

Project Reports.

**Describe the solution you'd like**

Be able to set email report reminder frequency or timings in Wagtail, including the email template.

**Priority**

- Medium (slowing you down)

**Affected roles**

- Staff
- Applicants

### PR DESCRIPTION
**Pull request by bickelj**
Fixes #3989

## Test Steps

Set up a fake SMTP server or use file-based email.

- Configure reports for projects in Wagtail Admin
- See that report reminders go out as configured

### CODE REVIEW

**Comment by bickelj** [comment]
@frjo Do you think this one is OK to merge?

**Comment by frjo** [comment]
Did a rebase and migration deconflict.

File: hypha/apply/projects/models/project.py
Comment (frjo) [inline hypha/apply/projects/models/project.py]: Title case in several places.

File: hypha/apply/projects/models/project.py
Comment (frjo) [inline hypha/apply/projects/models/project.py]: Can we rename this to something like "reminder_days"? My browser thinks this is a 2FA field.

File: hypha/apply/projects/models/project.py
Comment (frjo) [reply hypha/apply/projects/models/project.py]: Instead of "Project due date" I think "Report due date" makes it more clear. There is no place to set a "Project due date. We have `proposed_start` and `proposed_end` in the model but they are currently unused.

**Comment by frjo** [comment]
@bickelj Am I correct that this PR makes the following changes and additions to current functionality:

1. Move the remind before x days option from a command argument to Wagtail settings.
2. Allow to remind x days after or before report due date.
3. Allow to have multiple reminders with only one cron job.

**Review by frjo — CHANGES_REQUESTED** [review]

**Comment by bickelj** [comment]
@frjo said:
> Am I correct that this PR makes the following changes and additions to current functionality:
> 
>     1. Move the remind before x days option from a command argument to Wagtail settings.

I think the the setup defaulted to some days before and monthly, but you could vary the settings a bit on an individual project after creation. ~~So I don't think there was a way you could change that overall behavior. I don't know about a command argument.~~ I see what you mean about `hypha/apply/projects/management/commands/notify_report_due.py` but I haven't tried that.

>     2. Allow to remind x days after or before report due date.

Yes, and let these settings apply to all new projects.

>     3. Allow to have multiple reminders with only one cron job.

I think so, yes.

@frankduncan did I get that right?

**Comment by frjo** [comment]
@bickelj  If all titlecase can be changed to sentence case I can put this on test.

**Comment by bickelj** [comment]
@frjo I think I got all the places in the latest push, let me know if I didn't!

**Review by frjo — APPROVED** [review]

File: hypha/apply/projects/models/project.py
Comment (frjo) [inline hypha/apply/projects/models/project.py:545]: Can we add a help text here? something to explain that these settings are used when running `notify_report_due.py` as a cronjob. Please improve my draft below.

```
help_text=_("Set up a cronjob to run `notify_report_due.py`. That script will use the reminder settings below.")
```

**Review by frjo — CHANGES_REQUESTED** [review]
Would also be good to add some documentation around this.

Otherwise it seems to be working very nicely.

**Review by frjo — APPROVED** [review]

**Comment by frjo** [comment]
We have a migration numbering conflict with another PR on test but as soon as that is approved I will put this on test.

File: hypha/apply/projects/models/project.py
Comment (bickelj) [reply hypha/apply/projects/models/project.py:545]: I should look a little more closely at the command because I don't see `notify_report_due.py` in example cron jobs currently. OK, I see this command in the list from `python manage.py help`, so the note committed is probably OK as-is.
```
[projects]
    notify_report_due
```

**Comment by theskumar** [comment]
This PR look neat! 

@frjo @bickelj I think it would really help if we can add a how to guide in the docs on how to enable and setup the report reminders. 

Probably at https://docs.hypha.app/setup/administrators/setup-notifications/

**Comment by theskumar** [comment]
I don't think needs dedicated guide, but if it has to it will look something like this: https://docs.hypha.app/setup/administrators/setup-error-performance-monitoring/

**Comment by bickelj** [comment]
@theskumar I added some documentation in the most recent commit. Let me know what you think.

File: docs/setup/administrators/setup-notifications.md
Comment (bickelj) [inline docs/setup/administrators/setup-notifications.md:17]: I didn't want this to appear as if it were alike to "Email, Slack, ..." so I moved those into their own section before adding this one.

**Comment by bickelj** [comment]
@frjo @theskumar Can you take a look at the documentation and see if it looks OK? Do you think these report reminders can be merged in the next iteration?

**Comment by wes-otf** [comment]
I've got a test report setup to be due saturday so should be able to merge this tomorrow assuming all notifies right

**Review by wes-otf — APPROVED** [review]

**Comment by wes-otf** [comment]
worked well - thanks @bickelj!

**Pull request merged**
